src/main/java/org/openstreetmap/atlas/checks/validation/tag/ConstructionCheck.java

```diff
@@ -0,0 +1,234 @@
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        this.markAsFlagged(object.getOsmIdentifier());
+
+        final Map<String, String> tags = object.getTags();
+
+        final Optional<String> dateTag = getDateTag(tags);
+        if (dateTag.isPresent())
+        {
+            final String tagDate = tags.get(dateTag.get());
+
+            final Optional<LocalDate> parsedDate = parseDate(tagDate);
+            if (parsedDate.isPresent() && parsedDate.get().isBefore(TODAYS_DATE))
+            {
+                return Optional
+                        .of(createFlag(object, this.getLocalizedInstruction(0, dateTag.get())));
+            }
+        }
+
+        if (tags.containsKey("check_date"))
+        {
+            final Optional<LocalDate> parseDateChecked = parseDate(tags.get("check_date"));
+            if (parseDateChecked.isPresent())
+            {
+                final long monthsBetween = ChronoUnit.MONTHS.between(parseDateChecked.get(),
+                        TODAYS_DATE);
+                if (monthsBetween > this.oldCheckDateMonths)
+                {
+                    return Optional.of(createFlag(object,
+                            this.getLocalizedInstruction(1, this.oldCheckDateMonths)));
+                }
+            }
+        }
+
+        if (tags.containsKey("last_edit_time"))
+        {
+            final long timestamp = Long.parseLong(tags.get("last_edit_time"));
+            final LocalDate lastEditDate = Instant.ofEpochMilli(timestamp)
+                    .atZone(ZoneId.systemDefault()).toLocalDate();
+
+            final long numberOfDays = ChronoUnit.DAYS.between(lastEditDate, TODAYS_DATE);
+            if (numberOfDays > this.oldConstructionDays)
+            {
+                return Optional.of(createFlag(object,
+                        this.getLocalizedInstruction(2, this.oldConstructionDays)));
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Gets the tag that holds a date.
+     *
+     * @param keySet
+     *            Tags from the object.
+     * @return a tag that is considered a date for construction.
+     */
+    private Optional<String> getDateTag(final Map<String, String> keySet)
+    {
+        return DATE_TAGS.stream().filter(keySet::containsKey).findFirst();
+    }
+
+    /**
+     * Checks if the tags of an object signify it as being under construction.
+     *
+     * @param tags
+     *            Tags from the object
+     * @return true if the object is under construction, otherwise false
+     */
+    private boolean isConstruction(final Map<String, String> tags)
+    {
+        return tags.keySet().stream()
+                .anyMatch(tag -> tag.equals("construction")
```

Comment:
```suggestion
                .anyMatch(tag -> tag.equals(ConstructionTag.KEY.toString())
```
or if you pass in the atlas object:
```suggestion
                .anyMatch(tag -> Validators.hasValueFor(object, ConstructionTag.class)
```